apps/dashboard/src/lib/components/Course/components/Lesson/Exercises/NewExerciseModal.svelte

```diff
@@ -112,9 +127,26 @@
   //   }, 500);
   // }
 
-  onMount(async () => {
-    allTemplates = await getAllTemplates();
-  });
+  async function handleTemplateSelection() {
+    isTemplateFinishedLoading = true;
+    const template = allTemplates?.[selectedTag]?.find((t) => t.id === selectedTemplateId);
+
+    if (!template) return;
+
+    let fetchedTemplate: ExerciseTemplate;
+
+    try {
+      await templateApi.fetchTemplateById(template.id);
+      fetchedTemplate = templateApi.template[0];
```

Comment:
There shouldn't be state in this file, all state should be attached to the `templateApi`. Even the loading state should be from there `templateApi.isLoading`.

It already has a `isLoading` state, look at the `BaseApi` class

Reply:
fixed